BlockServer/core/devices_manager.py

```diff
@@ -0,0 +1,137 @@
+# This file is part of the ISIS IBEX application.
+# Copyright (C) 2012-2016 Science & Technology Facilities Council.
+# All rights reserved.
+#
+# This program is distributed in the hope that it will be useful.
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v1.0 which accompanies this distribution.
+# EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM
+# AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+#
+# You should have received a copy of the Eclipse Public License v1.0
+# along with this program; if not, you can obtain a copy from
+# https://www.eclipse.org/org/documents/epl-v10.php or
+# http://opensource.org/licenses/eclipse-1.0.php
+
+import os
+from server_common.utilities import print_and_log, compress_and_hex
+from BlockServer.core.file_path_manager import FILEPATH_MANAGER
+from BlockServer.fileIO.schema_checker import ConfigurationSchemaChecker
+from xml.dom import minidom
+
+GET_SCREENS = "GET_SCREENS"
+SET_SCREENS = "SET_SCREENS"
+
+SCREENS_FILE = "screens.xml"
+SCREENS_SCHEMA = "screens.xsd"
+
+
+class DevicesManager(object):
+    """Class for managing the PVs associated with devices"""
+    def __init__(self, block_server, cas, schema_folder, vc_manager):
+        """Constructor.
+
+        Args:
+            block_server (BlockServer): A reference to the BlockServer instance.
+            cas (CAServer): The channel access server for creating PVs on-the-fly
+            schema_folder (string): The filepath for the devices schema
+            vc_manager (ConfigVersionControl): The manager to allow version control modifications
+        """
+        self._schema_folder = schema_folder
+        self._cas = cas
+        self._devices_pvs = dict()
+        self._vc = vc_manager
+        self._bs = block_server
+        self._current_config = ""
+
+    def load_current(self):
+        """Create the PVs for all the devices found in the devices directory."""
+        devices_file_name = self.get_devices_filename()
```

Comment:
I think that if the file doesn't exist this returns blank and an exception will be thrown and caught about a file with no file name does not exist